File: app/src/main/java/info/horriblesubs/sher/ui/b/Explore.java
package info.horriblesubs.sher.ui.b;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

import org.jetbrains.annotations.NotNull;

import info.horriblesubs.sher.AppMe;
import info.horriblesubs.sher.R;
import info.horriblesubs.sher.common.FragmentRefresh;
import info.horriblesubs.sher.ui.a.Ads;
import info.horriblesubs.sher.ui.a.Menu;

public class Explore extends AppCompatActivity implements View.OnClickListener, SwipeRefreshLayout.OnRefreshListener {

    private SwipeRefreshLayout refreshLayout;
    private Menu menu;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        AppMe.appMe.setTheme();
        setContentView(R.layout.b_activity_0);
        refreshLayout = findViewById(R.id.swipeRefreshLayout);
        findViewById(R.id.fab).setOnClickListener(this);
        refreshLayout.setOnRefreshListener(this);
        Ads.BannerAd.load(this);
        menu = Menu.all();
    }

    public void onEndRefresh() {
        refreshLayout.setRefreshing(false);
    }

    @Deprecated
    public boolean onMenuItemClick(@NotNull MenuItem item) {
        switch (item.getItemId()) {
            case R.id.refresh:
                Fragment fragment = getSupportFragmentManager().findFragmentById(R.id.fragment3);
                if (fragment instanceof FragmentRefresh)
                    ((FragmentRefresh) fragment).onRefresh();
                return true;
            case R.id.browser:
                try {
                    String link = getResources().getString(R.string.horrible_subs_url);
                    Intent intent = new Intent(Intent.ACTION_VIEW);
                    intent.addCategory(Intent.CATEGORY_BROWSABLE);
                    intent.setData(Uri.parse(link));
                    startActivity(intent);
                } catch (Exception e) {
                    Toast.makeText(this, getResources().getString(R.string.app_not_available), Toast.LENGTH_SHORT).show();
                    e.printStackTrace();
                }
                return true;
        }
        return false;
    }

    @Override
    public void onClick(@NotNull View view) {
        if (view.getId() == R.id.fab)
            menu.show(getSupportFragmentManager());
    }

    @Override
    public void onRefresh() {
        Fragment fragment = getSupportFragmentManager().findFragmentById(R.id.fragment3);
        if (fragment instanceof FragmentRefresh) ((FragmentRefresh) fragment).onRefresh();
        fragment = getSupportFragmentManager().findFragmentById(R.id.fragment1);
        if (fragment instanceof FragmentRefresh) ((FragmentRefresh) fragment).onRefresh();
    }
}

File: app/src/main/java/info/horriblesubs/sher/ui/main/explore/Explore.kt
package info.horriblesubs.sher.ui.main.explore

import android.os.Bundle
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import android.widget.PopupMenu
import androidx.annotation.IdRes
import androidx.fragment.app.Fragment
import info.horriblesubs.sher.R
import info.horriblesubs.sher.common.MenuHandler
import info.horriblesubs.sher.ui.main.explore.Explore.Keys.*
import info.horriblesubs.sher.ui.main.explore.random.Random
import info.horriblesubs.sher.ui.main.explore.recent.Recent
import info.horriblesubs.sher.ui.main.explore.trending.Trending

class Explore: Fragment(), PopupMenu.OnMenuItemClickListener {
    private var menuHandler: MenuHandler? = null

    override fun onCreateView(inflater: LayoutInflater, group: ViewGroup?, bundle: Bundle?): View? {
        return inflater.inflate(R.layout._b_fragment_1, group, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        menuHandler = MenuHandler(this, view, R.menu.menu_a)
        fragment(R.id.fragment1, TRENDING)
        fragment(R.id.fragment2, RANDOM)
        fragment(R.id.fragment3, RECENT)
    }

    override fun onMenuItemClick(item: MenuItem?): Boolean {
        (childFragmentManager.findFragmentById(R.id.fragment1) as Trending?)?.onMenuItemClick(item)
        (childFragmentManager.findFragmentById(R.id.fragment3) as Recent?)?.onMenuItemClick(item)
        return true
    }

    private fun fragment(@IdRes id: Int, key: Keys) {
        childFragmentManager.beginTransaction().replace(id, key.fragment, key.name).commit()
    }

    private enum class Keys {
        RECENT, RANDOM, TRENDING;
        internal val fragment: Fragment
            get() = when (this) {
                TRENDING -> Trending()
                RECENT -> Recent()
                RANDOM -> Random()
            }
    }
}